src/platform/storage_winkernel.c

```diff
@@ -362,21 +437,26 @@ CxPlatStorageOpen(
 _IRQL_requires_max_(PASSIVE_LEVEL)
 void
 CxPlatStorageClose(
-    _In_opt_ CXPLAT_STORAGE* Storage
+    _In_opt_ _Post_invalid_ CXPLAT_STORAGE* Storage
     )
 {
     if (Storage != NULL) {
         CXPLAT_EVENT CleanupEvent;
-        CxPlatEventInitialize(&CleanupEvent, TRUE, FALSE);
+        if (Storage->Callback != NULL) {
+            CxPlatEventInitialize(&CleanupEvent, TRUE, FALSE);
 
-        CxPlatLockAcquire(&Storage->Lock);
+            CxPlatLockAcquire(&Storage->Lock);
```

Comment:
nit: Because the locking logic now because slightly more complex (conditional), it could be a good time to add a comment on the lock definition to explain what it protects (and I imagine you have it in mind right now)